### ISSUE TITLE
Backup to self-hosted server

### ISSUE THREAD

**Issue by chaganiu** [raivo-otp/ios-application#49]
**Is your feature request related to a problem? Please describe.**
Currently we have 3 backup options:
* none
* iCloud
* local backup (coming soon)

It would be nice to have and option to backup/restore from a self-hosted server.  

**Describe the solution you'd like**
An option to backup to a self-hosted server

**Describe alternatives you've considered**
local backup when it is available but that seems like a manual process to store on a server i control.

**Comment by tijme** [comment]
Good idea. What kind of protocol would you prefer to use for backup to a self-hosted server?

**Comment by chaganiu** [comment]
Honestly, i'm not sure.  It could be as straightforward as backing up to a nextcloud instance.  whatever is easiest.  As long as the backup is encrypted locally with the master password I think anything would be fine.  For me personally, being able to send an encrypted backup to my nextcloud instance on a periodic basis would suffice (assuming restoring would be as simple as getting the backup from my nextcloud instance and loading it into this fantastic app).

**Comment by chris-aeviator** [comment]
nextcloud uses webdav

**Comment by Rustymage** [comment]
I would say either the SFTP or WebDAV protocol gives the most flexibility to those that want to backup outside of iCloud.

**Comment by cat0x1f** [comment]
i have a idea:
save the database as json (or other text file), and it can be saved on onedrive, sftp, and more.
Sync the database by Onedrive, others can develop clients on other platforms easily.
